### ISSUE TITLE
Improve (?) printedkey produced by NormalizeNameAndKey

### ISSUE THREAD

**Issue by fingolfin** [frankluebeck/GAPDoc#42]
Consider this input:
```
gap> a:=rec( From := rec( BibTeX := true ), Label := "MR3163022", Type := "article",
>   author := "Álvarez, Amelia and Bravo, José Luis and Christopher, Colin",
>   doi := "10.1017/etds.2012.143", fjournal := "Ergodic Theory and Dynamical Systems",
>   issn := "0143-3857", journal := "Ergodic Theory Dynam. Systems", mrclass := "34C05 (34C25)",
>   mrnumber := "3163022", mrreviewer := "Y. Yomdin", number := "1", pages := "1–20",
>   title := "On the trigonometric moment problem", url := "https://doi.org/10.1017/etds.2012.143",
>   volume := "34", year := "2014" );;
gap> NormalizeNameAndKey(a);
gap> a.printedkey;
"XBC14"
```
So the Unicode character `Á` is converted to `X` for the printedkey. At least for latin characters like these, I'd think that perhaps converting it to A instead would result in a nicer output? Coincidentally, that's what I saw before, when the author field was still encoded using LaTeX notation: `author := "\\'{A}lvarez, Amelia...`

**Comment by fingolfin** [comment]
Another somewhat odd one is this:
```
gap> a:=ShallowCopy(First(bib[1], a -> a.title = """Computing {G}r\"{o}bner bases associated with lattices"""));
rec( From := rec( BibTeX := true ), Label := "MR3569967", Type := "article",
  author := "Álvarez-Barrientos, Ismara and Borges-Quintana, Mijail and\n              Borges-Tren\
ard, Miguel Angel and Panario, Daniel", doi := "10.3934/amc.2016045",
  fjournal := "Advances in Mathematics of Communications", issn := "1930-5346",
  journal := "Adv. Math. Commun.", mrclass := "94B25", mrnumber := "3569967",
  mrreviewer := "Guangzhou Chen", number := "4", pages := "851–860",
  title := "Computing {G}r\\\"{o}bner bases associated with lattices",
  url := "https://doi.org/10.3934/amc.2016045", volume := "10", year := "2016" )
gap> NormalizeNameAndKey(a);
gap> a.printedkey;
"BB+16"
```
In the past (before the improvements in 753f8cbd035db1401ca787256254b8e582db8e05) we actually got the key `AB+16` here (as `Álvarez` was `\'{A}lvarez` then).

**Comment by frankluebeck** [comment]
Thanks for these examples. Indeed, if names in BibTeX are given in UTF-8 encoding it is better to compute a simplified ASCII version before computing the `.printedkey` and `.keylong` fields. Fortunately, GAPDoc has a utility `SimplifiedUnicodeString` to do this.

The behaviour is changed by commit "Better keys in BibTeX for non-ASCII initials" (5a0ed8c). Now it seems to work nicely with the names given in LaTeX or in UTF-8.

**Comment by alex-konovalov** [comment]
Thanks @frankluebeck! When this will be in the released GAPDoc, this will help with GAP bibliography pages.

**Comment by fingolfin** [comment]
Indeed, I noticed this while working on the GAP bibliography pages; when I created https://github.com/gap-system/GapWWW/pull/204 I noticed a lots of diffs I didn't understand, until I realized I run `convbib.g` with the latest GAPDoc development version... I then re-run with GAP 4.11.0 and the last released GAPDoc, just to keep the diff smaller. My plan is to re-run it now with the fix Frank just made.

**Comment by frankluebeck** [comment]
Ok, tell me if you are content with the results. If yes, I will prepare a GAPDoc release that contains the recent changes.

**Comment by fingolfin** [comment]
@frankluebeck We are happy with the result and would appreciate a new GAPDoc release with them! Then we can hopefully release it together with GAP 4.11.1 in the not so distant future...

**Comment by frankluebeck** [comment]
I was not yet happy with the current file `gap-publishednicer.bib` on GAPs website. I have again added more translations from LaTeX to UTF-8 in GAPDoc.

I suggest to regenerate this file again with the current GAPDoc which is now released as version 1.6.4.

**Comment by alex-konovalov** [comment]
@frankluebeck thanks, it will be regenerated when someone (me?) will be updating the GAP bibliography next time.
